motr/ut/idx_dix.c

```diff
@@ -1276,15 +1291,34 @@ static void st_dtm0_r_common(uint32_t sdev_id)
 		return;
 
 	idx_setup();
+	/** PUT redo */
 	exec_one_by_one(1, M0_IC_PUT);
-	dtm0_ut_send_redo(&duc.duc_ifid, sdev_id, &key, &val);
+	dtm0_ut_send_redo(&duc.duc_ifid, sdev_id, &key, &val,
+			  M0_CAS_PUT_FOP_OPCODE);
+
+	/* XXX dirty hack, but now we don't have completion notification */
+	rem = 2ULL * M0_TIME_ONE_SECOND;
+        while (rem != 0)
+                m0_nanosleep(rem, &rem);
+
+	dtm0_ut_read_and_check(key, val, M0_CAS_PUT_FOP_OPCODE);
+
+	idx_teardown();
```

Comment:
Please remove lines from 1306 to 1312. Currently test for DEL does nothing since no records exist on a CAS side. So the test can be:
- 1 PUT (just to have something else but the record sent using REDO), existing code
- REDO PUT, existing code
- Read and check (val expected), existing code
- REDO DEL, new code
- Read and check (-ENOENT expected), new code